katharsis-client/src/main/java/io/katharsis/client/internal/ErrorResponseDeserializer.java

```diff
@@ -0,0 +1,117 @@
+package io.katharsis.client.internal;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import io.katharsis.errorhandling.ErrorData;
+import io.katharsis.errorhandling.ErrorResponse;
+
+/**
+ * Serializes top-level Errors object.
+ */
+public class ErrorResponseDeserializer extends JsonDeserializer<ErrorResponse> {
+
+	private static final String LINKS = "links";
+	private static final String ID = "id";
+	private static final String ABOUT_LINK = "about";
+	private static final String STATUS = "status";
+	private static final String CODE = "code";
+	private static final String TITLE = "title";
+	private static final String DETAIL = "detail";
+	private static final String SOURCE = "source";
+	private static final String POINTER = "pointer";
+	private static final String PARAMETER = "parameter";
+	private static final String META = "meta";
+
+	@Override
+	public ErrorResponse deserialize(JsonParser jp, DeserializationContext context)
+			throws IOException, JsonProcessingException {
+
+		JsonNode node = jp.readValueAsTree();
+		if (node == null) {
+			return null;
+		}
+
+		JsonNode errorsNode = node.get(ErrorResponse.ERRORS);
+		List<ErrorData> errors = new ArrayList<ErrorData>();
+		if (errorsNode != null) {
+			if (!errorsNode.isArray()) {
+				throw new RuntimeException("data field has wrong type: " + node.toString());
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)